AdjustedSurvivalCurves/adjcurves_plain.Rmd

```diff
@@ -0,0 +1,186 @@
+
+```{r, echo=T, message=F, warning=F}
+# Required packages
+library(tidyverse)
+library(survey)
+library(survival)
+library(gtsummary)
+library(survminer)
+```
+
+```{r, echo=F, message=F, warning=F, results='asis'}
+set.seed(1)
+n <- 10000
+
+# 60% percent women
+female <- rbinom(n, 1, 0.6)
+# Men has the enzym with a prob of 0.3; women have
+# a 5 times higher chance of having the enzym
+enzym <- ifelse(runif(n) < plogis(qlogis(0.3)+log(5)*female), 1, 0)
+# Those with enzym receive medication with prob 0.75
+medi <- ifelse(runif(n) < plogis(log(3)*enzym), 1, 0)
+# Hazard of dying: HR=10 of those with enyzm; No effect for medication
+haz <- 0.01*exp(log(10)*enzym)
+# Time to death
+time_to_death <- -log(runif(n))/haz
+# Censoring time: Mean duration 20 days
+censored <- rweibull(n, 1, 20)
+# Follow-up time
+fup <- pmin(time_to_death, censored)
+# Death
+death <- ifelse(time_to_death<=censored, 1, 0)
+# Data
+data <- data.frame(fup, death, medi, enzym, female)
+```
+
+## Descriptive table
+
+The table below shows a descriptive summary of the study population, by medication.
+
+```{r, echo=T, message=F, warning=F}
+data %>% tbl_summary(by=medi, label=list(medi ~ "Medication")) %>% as_kable() 
+```
+
+## Unadjusted Kaplan-Meier curve and Cox-modelling
+
+A naive (unadjusted) survival analysis of the data reveals the following Kaplan-Meier plot. We conclude that the medication has an effect on survival.
+
+```{r, echo=T, message=F, warning=F}
+mod <- survfit(Surv(fup, death)~medi, data=data)
+ggsurvplot(mod, data=data, palette=c("#CC0000", "black"), censor=F)
+```
+
+```{r, echo=T, message=F, warning=F}
+mod_cox_unadjusted <- coxph(Surv(fup, death) ~ medi, data = data)
+mod_cox_unadjusted
+```
+
+An unadjusted Cox proportional hazard model shows that patients with medication have `r round(exp(mod_cox_unadjusted$coef[1]), 1)` higher hazard of death compared to those without medication.
+
+```{r, echo=T, message=F, warning=F}
+mod_cox_adjusted <- coxph(Surv(fup, death) ~ medi+enzym+female, data = data)
+mod_cox_adjusted
+```
+
+What happens if we adjust for enzym and sex? Then the effect of the medication on death vanish (hazard ratio=`r round(exp(mod_cox_adjusted$coef[1]), 2)`).
+
+## IPW modelling
+
+An IPW modelling approach construct treatment weights (here medication) given known covariates (here sex and enzym) using a logistic regression model.
+
+```{r, echo=T, message=F, warning=F, results='markup'}
+# IPW denominator
+mod <- glm(medi ~ female+enzym, data=data, family=binomial())
+
+data$ipw <- NA
+# Probabilty of treatment
+data$ipw <- predict(mod, data=data, type="response")
+# Probabilty of non-treatment
+data$ipw[data$medi==0] <- 1-predict(mod, data=data, type="response")[data$medi==0]
+```
+
+We construct stabilized weights, since they can provide narrower confidence intervals (@hernan_book).
```

Comment:
I am not a big expert here, but I thought that the un-stabilized weights might actually produce too narrow CIs for the estimates, as they artificially blow up the sample size of the pseudopopulation, while stabilized weights spread around 1, and therefore, better represent the size of the original data set.